src/java/org/apache/ivy/core/cache/DefaultRepositoryCacheManager.java

```diff
@@ -1215,26 +1217,22 @@ public ResolvedResource resolve(Artifact artifact) {
                     for (int i = 0; i < confs.length; i++) {
                         Artifact[] arts = md.getArtifacts(confs[i]);
                         for (int j = 0; j < arts.length; j++) {
-                            Artifact transformedArtifact = NameSpaceHelper.transform(
-                                arts[j], options.getNamespace().getToSystemTransformer());
-                            ArtifactOrigin origin = getSavedArtifactOrigin(
-                                transformedArtifact);
-                            File artFile = getArchiveFileInCache(
-                                transformedArtifact, origin, false);
-                            if (artFile.exists()) {
-                                Message.debug("deleting " + artFile);
-                                if (!artFile.delete()) {
-                                    // Old artifacts couldn't get deleted!
-                                    // Restore the original ivy file so the next time we
-                                    // resolve the old artifacts are deleted again
-                                    backupDownloader.restore();
-                                    Message.error("Couldn't delete outdated artifact from cache: " + artFile);
-                                    return null;
-                                }
-                            }
-                            removeSavedArtifactOrigin(transformedArtifact);
+                            if (!prepAndDeleteArtifact(arts[j], options, backupDownloader)) return null;
                         }
                     }
+
+                    final Artifact sourceArtifact = new MDArtifact(md, mrid.getName(), "src", "jar", null,
+                        Collections.singletonMap("m:classifier", "sources"));
+
+                    final Artifact srcArtifact = new MDArtifact(md, mrid.getName(), "src", "jar", null,
+                        Collections.singletonMap("m:classifier", "src"));
+
+                    final Artifact javadocArtifact = new MDArtifact(md, mrid.getName(), "doc", "jar", null,
+                        Collections.singletonMap("m:classifier", "javadoc"));
```

Comment:
I'm explicitly specifying the Artifact details here instead using methods on `PomModuleDescriptorBuilder` (which I was doing in [a previous attempt](https://github.com/sbt/ivy/pull/15/files#diff-256c5c36f5c173caa043a45c357ebed5R1224)) because that sets the type to `"source"` and `"javaodc"` which makes it look artifacts at

```
~/.ivy2/cache/com.dwijnand/sbt-gh1750-bippy_2.11/sources/sbt-gh1750-bippy_2.11-0.1.0-SNAPSHOT-sources.jar
~/.ivy2/cache/com.dwijnand/sbt-gh1750-bippy_2.11/javadocs/sbt-gh1750-bippy_2.11-0.1.0-SNAPSHOT-javadoc.jar
```

rather than

```
~/.ivy2/cache/com.dwijnand/sbt-gh1750-bippy_2.11/srcs/sbt-gh1750-bippy_2.11-0.1.0-SNAPSHOT-sources.jar
~/.ivy2/cache/com.dwijnand/sbt-gh1750-bippy_2.11/docs/sbt-gh1750-bippy_2.11-0.1.0-SNAPSHOT-javadoc.jar
```

given the ivy/artifact pattern

```
(scala_[scalaVersion]/)(sbt_[sbtVersion]/)[organisation]/[module](/[branch])/[type]s/[artifact]-[revision](-[classifier])(.[ext])
```

I then found this [correspondence map](https://github.com/sbt/sbt/blob/v0.13.9/ivy/src/main/scala/sbt/Artifact.scala#L66) in sbt, so decided to just define the types here as `"src"` and `"doc"` instead. Let me know if I fixing this some other way..